rama-http/src/service/web/endpoint/response/octet_stream.rs

```diff
@@ -0,0 +1,174 @@
+use super::{Headers, IntoResponse};
+use crate::headers::ContentType;
+use crate::{Body, Response, StatusCode};
+use rama_http_headers::{ContentDisposition, ContentRange, Error, HeaderMapExt};
+use std::fmt;
+use std::ops::RangeBounds;
+
+/// An octet-stream response for serving arbitrary binary data.
+///
+/// Will automatically get `Content-Type: application/octet-stream`.
+///
+/// This is commonly used for:
+/// - Serving unknown or arbitrary binary files
+/// - Downloadable content that doesn't fit other MIME types
+/// - Raw binary data streams
+///
+/// # Examples
+///
+/// ## Basic binary response
+///
+/// ```
+/// use rama_http::service::web::response::{IntoResponse, OctetStream};
+///
+/// async fn handler() -> impl IntoResponse {
+///     let data = vec![0x48, 0x65, 0x6C, 0x6C, 0x6F]; // "Hello"
+///     OctetStream::new(data)
+/// }
+/// ```
+///
+/// ## Download with filename
+///
+/// ```
+/// use rama_http::service::web::response::{IntoResponse, OctetStream};
+///
+/// async fn download() -> impl IntoResponse {
+///     let data = b"file contents".to_vec();
+///     OctetStream::new(data).with_file_name("data.bin".to_string())
+/// }
+/// ```
+///
+pub struct OctetStream<T> {
+    data: T,
```

Comment:
This is. to be a `stream`, not data. Same same, just different name. But the trait bound below for it also has to change. More in other comment.

Next to that it is important to also add an optional `content_size: Option<usize>` property as not all streams will have this information. Next to that you can also provide a setter to optionally set the content-size.

Would also be nice if you can provide an alternative (extra) constructor to easily create an `OctetStream` from a file on disk, something like:

```
pub async fn try_from_path(path: impl AsRef<Path>) -> io::Result<OctetStream<ReaderStream<File>>>
```

This will provide the content_size and file name in case it is available from the metadata (and can be read, be graceful here)